### ISSUE TITLE
SDPA operation fails with head dim not divisible by 32

### ISSUE THREAD

**Issue by jserbedzijaTT** [tenstorrent/tt-metal#33434]
### Describe the bug

The scaled_dot_product_attention operation fails when query and key/value head dim is not divisible by 32

### Steps to reproduce the issue

Run the following test script:
```py
# SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
# SPDX-License-Identifier: Apache-2.0

import pytest
import torch
import ttnn


@pytest.mark.parametrize(
    "q_shape,k_shape,v_shape,mask_shape",
    [
        # head_dim=48 (not divisible by 32)
        ((1, 8, 64, 48), (1, 8, 64, 48), (1, 8, 64, 48), (1, 1, 64, 64)),
        # head_dim=80 (not divisible by 32)
        ((1, 4, 128, 80), (1, 4, 128, 80), (1, 4, 128, 80), (1, 1, 128, 128)),
    ],
)
def test_sdpa_non_divisible_head_dim(q_shape, k_shape, v_shape, mask_shape):
    """Test SDPA with shapes that have head_dim not divisible by 32."""

    torch.manual_seed(42)

    Q = torch.randn(q_shape)
    K = torch.randn(k_shape)
    V = torch.randn(v_shape)
    attention_mask = torch.zeros(mask_shape)

    device = ttnn.open_device(device_id=0)

    try:
        tt_Q = ttnn.from_torch(Q, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
        tt_K = ttnn.from_torch(K, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
        tt_V = ttnn.from_torch(V, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
        tt_attention_mask = ttnn.from_torch(attention_mask, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)

        tt_output = ttnn.transformer.scaled_dot_product_attention(
            tt_Q,
            tt_K,
            tt_V,
            is_causal=False,
            attn_mask=tt_attention_mask,
        )

        output = ttnn.to_torch(tt_output)

    finally:
        ttnn.close_device(device)

```

### Expected behavior

Support padding the head dim to be divisible by 32 and then slice the output back to the original dimensions.

**Comment by skhatriTT** [comment]
Hey @cglagovichTT, can you please provide a guesstimate on when is this scheduled for? this will help us plan

**Comment by jserbedzijaTT** [comment]
Any update on this? @cglagovichTT

**Comment by cglagovichTT** [comment]
Hi, I'm sorry but this is unscheduled. Team's bandwidth to address is low.
I would appreciate if Forge team could assist.
